src/cdomains/apron/affineEqualityDomain.apron.ml

```diff
@@ -0,0 +1,725 @@
+open Prelude
+open Pretty
+open Cil
+module M = Messages
+open Apron
+
+
+module type Arith =
+sig
+  type t
+  val add : t -> t -> t
+  val sub : t -> t -> t
+  val mul : t -> t -> t
+  val div : t -> t -> t
+  val neg : t -> t
+  val abs : t -> t
+  val cmp : t -> t -> int
+  val to_string:  t -> string
+  val of_int: int -> t
+end
```

Comment:
Why this? There's already `IntOps` for this exact reason to abstract over integer types.

Reply:
`IntOps` does unfortunately not include abstractions for rationals. The 'smallest' interface inside `IntOps`,` IntOpsBase`, would require shifting operations etc. which don't work for rationals.

Reply:
Confusingly, nothing in this interface implies nor requires the use of rationals to implement it though. There's just `of_int`, which suggests that integers would suffice and indeed every `IntOps` implementation also satisfies this interface (other than `compare` having the non-standard name `cmp`). What would break down if one were to try doing that?

Reply:
Ok that was true until my latest commit 6654acd where I added the `get_den` and `get_num` functions to the interface. Now it should be clearly distinct from `IntOps`

Reply:
I'm wondering of something like `RatOps` would be clearer that it's about rationals, not just any arithmetic.

Reply:
Yes, that's indeed clearer. Renamed in a50ac6